Commented out ingame spawn error, due to torch not having (all) spawnpoints

Crow-Control opened the pull request:
There currently aren't really other spawnpoints to choose from, so I … disabled all options except cryostorage and commented out the error. They are still there just commented out if need be.

Just a short temporary fix, till there is more clearity about spawn locations and options, but at least we dont have the error when joining anymore this way and players don't get non-active options in config.

Its quick and dirty, but others can build still build upon it, or not, for future spawn improvements.

Commit:
There currently aren't really other spawnpoints to choose from, so I disabled all choises except cryostorage and commented out the error.
They are still there just commented out if need be.

MistakeNot4892 commented on code/game/jobs/job_controller.dm:
```diff
@@ -603,7 +603,7 @@ var/global/datum/controller/occupations/job_master
 	if(C.prefs.spawnpoint)
 		if(!(C.prefs.spawnpoint in using_map.allowed_spawns))
 			if(H)
```
Might as well comment out the now-empty if too?

Crow-Control replied:
might as well indeed!
Less unneeded active code the better... Thanks for the tip!

Crow-Control replied:
wont it error that way? after the error it continues setting the spawnpoint to NULL.

Crow-Control replied:
After the Ingame error message it sets the spawnpoint to NULL I mean... so leaving that out and continuing with an invalid spawnpoint set, wont that give an error?

Crow-Control replied:
damn got it, sorry was misreading which if you meant.

MistakeNot4892 replied:
Haha no worries.

Commit:
indentation (+2 squashed commit)

Squashed commit:

[0cd7ca1f0c] Old maps break when removing config options. readding config while removing in game error message

[289cc34234] As requested, fixed the changelog to reflext the changed PR goals. (+1 squashed commits)

Squashed commits:

[0f6ca3b7f7] Added the if above the ingame error to the commented out section. (+1 squashed commits)

Squashed commits:

[c89fa4b9c7] Added a changelog

atlantiscze commented:
:-1: This PR conceptually breaks the Exodus map's spawning system. Instead of removing these spawn options, hide/disable them only selectively depending on which map is being used.

Crow-Control commented:
@atlantiscze I dont really agree, are we going to keep all legency code for that reason?

Crow-Control commented:
But @atlantiscze , i got your point will just remove the error from now. Seems not really needed as user error anyway.

PsiOmegaDelta commented:
It ensures we convert legacy code into something that's multi-map compatible.

Crow-Control commented:
I understand, so i have dropped the code concerning spawn choices and only removed the Client side error on join.

I think we can agree the user has no need for that warning at the moment and the possible future.

Datraen commented:
:-1: It's better to properly fix the problem rather than hiding it.

Crow-Control commented:
Its meant as a temporary fix rater than a permanent one.
Easy code, quick commit.

I do not have the internal code structure knowlage to create an "map selection"-if

Asanadas commented:
Can you change / modify the changelog and then squash?

Crow-Control commented:
squash?
I am not using git for that long and do not like to experiment with features in a public PR :s

MistakeNot4892 commented:
Squashing is a git function that collapses several commits into a single commit. It's useful for reducing the number of commits you are adding to the git history of another repository when you do a PR.

http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

atlantiscze commented:
@Ornias1993 In general, these temporary fixes are something i would recommend to avoid. They only hide the problem, and it may (read: it usually does) come back again when an unrelated change is done.

I strongly advise using actual git client, or a GUI client that works with git (my personal preference is SourceTree) and a desktop copy of the code. Using web editor is the worst thing you can do as you have no way to test that your changes actually work. This applies even more when touching sensitive things, like spawning.

In this particular case, you should add a list variable to using_map datum, that would list available spawn points (if there's not one already). Then you can compare against this list, instead of using a hard-coded of spawn points.

Crow-Control commented:
@atlantiscze 
1. I do use sourcetree (the real gui GIT client sucks balls)
2. Ofcourse I use a desktop copy and no, i make my own choices how i do thinks and why.
2.1 Yes I should compile more beforehand. Sorry.
2.2 No I do not approve of spending coding time on legency support and I will not do that.
3. I like to do easy fixes unless it is an issue that really needs more code to solve.

Ergo:
If you do not like what I PR, reject it.
If you like: use it
If you have feedback: Thank you.

I understand that you want me to write more, I don't want to for this problem. This was a small annoyance and a small fix of mine. 

I rater spend my time on projects that infact are needed (and not just for legency support), such as my PR for the Oxygen_pump, that one really needs fixing and not like this just for the sake of "neat" and legencysupport.

@Zuhayr I got it working in sourcetree. Seems sourcetree has a...uhmm... "unique" way of handeling squashing what seems to be the issue why I never got it to work in the first place.

(and I personally are agains squashing, I like the way GIT logs small edits and changes)

PsiOmegaDelta commented:
So much controversy for such a minor thing.

I'm sufficiently alright with the proposed change in its current state, only the code should be removed not commented out. That's what version control is for.
Yes, the error message is nice to have but at the same time it's a little annoying when we have those invalid spawn points.

I'll again address the legacy point with more words. Baystation12 is the upstream repository for a fair number of downstream servers, out of which a number still run the old Exodus map or even other non-Torch alternatives.
As such we should strive towards making adjustments which work for any and all maps and avoid having them be Torch-specific.

Crow-Control commented:
@PsiOmegaDelta I understand thank you for clearing it up.
I will fix it by deleting that code for the moment today, as requested.

I would like to point out that an error would be nice at some time, but maybe not user-sided?

Commit:
removing error code completely, as requested by @PsiOmegaDelta

comma commented:
I tihnk better way to handle this would be to sanitize the char slot on load. I think it was left as is beforehand because slots were shared between maps, but now they are map-specific.
On sanitizing, just make it pick fist available spawn method. Only list map available methods in char setup too.

atlantiscze commented:
I'll be opening a PR that actually fixes the problem shortly.

Crow-Control commented:
Thanks @atlantiscze :D

Crow-Control commented:
Closed due to beter solution by @atlantiscze  in #17106

atlantiscze commented:
Always glad to help ^^
